StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/STPElementsSession.swift

```diff
@@ -31,6 +31,12 @@ final class STPElementsSession: NSObject {
     /// A map describing payment method types form specs.
     let paymentMethodSpecs: [[AnyHashable: Any]]?
 
+    /// An object containing Customer's saved payment methods information
+    let legacy_customer: STPLegacyElementsCustomer?
+
+    /// An error associated with operations surrounding legacy_customer
+    let customer_error: STPElementsCustomerError?
```

Comment:
I went down the path of decoding to a `String` -- I got this working, but then we run into an issue when we try to throw an error:
```
        if let customer_error = elementSession.customerError {
            throw CustomerSheetError.errorFetchingSavedPaymentMethods(customer_error)
        }
```
since `customer_error` is a `String`, and the associated type is expecting a type of Error, we get a compilation error.

Seems okay just to keep this error type?

Reply:
Ah I didn't see it was an `Error`.  I think it's still better to do `let customerError: Error?` and add a case to `PaymentSheetError` create a whole new class, but I won't block on it.

Reply:
Chatted offline -- attempted to deserialize to a new enum value within PaymentSheetError, however, there is another case where we are using `STPElementsCustomerError.decodeObject` which seems to make me think that we should keep `STPElementsCustomerError`. In `PaymentSheetLoader`,  there is `getPaymentMethodsFrom`, where we are looking at `.allResponseFields` to detect if there is an error in an already deserialized payload.

Based on this use case, seems like we should keep `STPElementsCustomerError`

Reply:
I'm not sure the difficulties there are related to what type we use to represent the error (this class vs an enum case).  

But also, I don't see why that method needs to decode the error in the first place.  If there was an error, elementsCustomerInformation.paymentMethods is nil. I didn't notice this in my earlier pass, but can you simplify 
```
        case .paymentIntent(let underlyingIntent):
            let dict = underlyingIntent.allResponseFields
            if STPElementsCustomerError.decodedObject(fromAPIResponse: dict) != nil {
                return nil
            } else if let elementsCustomerInformation = STPElementsCustomerInformation.decodedObject(fromAPIResponse: dict) {
                return elementsCustomerInformation.paymentMethods
            }
```

and rewrite like:

```
        case .paymentIntent(let underlyingIntent):
            return STPElementsCustomerInformation.decodedObject(fromAPIResponse: underlyingIntent.allResponseFields)?.paymentMethods
        }
```